### PR TITLE
dts: ite: Move common it8801 configurations to a shared include file

### PR DESCRIPTION
**Pull request by GTLin08**
Extracted common it8801 configuration into a new it8801-common-cfg.dtsi.

### CODE REVIEW

File: dts/riscv/ite/it8801-common-cfg.dtsi [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ite/it8801-mfd-map.dtsi>
+
+&it8801_i2c {
+	it8801_mfd: it8801@38 {
+		compatible = "ite,it8801-mfd";
+		reg = <0x38>;
+		irq-gpios = <&gpioa 1 0>;
```
Comment (fabiobaltieri) [inline dts/riscv/ite/it8801-common-cfg.dtsi:13]: this should probably not be here? I'd imagine you can connect it to any pin

File: tests/drivers/build_all/input/it82xx2_evb.overlay [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+				       17 18 16 15 11 12>;
+			row-size = <8>;
+			col-size = <13>;
+
+			kscan_input: kscan-input {
+				compatible = "zephyr,kscan-input";
+			};
+		};
+	};
+};
```
Comment (fabiobaltieri) [inline tests/drivers/build_all/input/it82xx2_evb.overlay:37]: no need to replicate the whole lot, just the overridden properties, in this case I think

```
&it8801_mfd {
        irq-gpios = <&gpioa 1 0>;

        ioex_it8801_kbd: it8801_kbd@40  {
                status = "okay";
                mfdctrl = <&kso18_gp01_default &kso20_gp23_default>;
                kso-mapping = <0 1 20 3 4 5 6 17 18 16 15 11 12>;
                row-size = <8>;
                col-size = <13>;

                kscan_input: kscan-input {
                        compatible = "zephyr,kscan-input";
                };
        };
};
```

same for the other tests

File: dts/riscv/ite/it8801-common-cfg.dtsi (hunk 1 above)
Comment (GTLin08) [reply dts/riscv/ite/it8801-common-cfg.dtsi]: Removed.

File: tests/drivers/build_all/input/it82xx2_evb.overlay (hunk 2 above)
Comment (GTLin08) [reply tests/drivers/build_all/input/it82xx2_evb.overlay]: Done.

**Review by fabiobaltieri — APPROVED** [review]
cool

**Review by kartben — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**